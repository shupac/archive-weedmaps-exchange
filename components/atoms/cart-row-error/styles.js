import styled from 'styled-components';
import { WmTheme } from '@ghostgroup/ui';
import theme from 'lib/styles/theme';
import { rem } from 'polished';

const { border, state } = WmTheme.style;

const AlertDanger = styled.div`
  width: 100%;
  border: 1px solid ${border.error};
  border-radius: 3px;
  background-color: ${theme.colors.bg.error};
  margin: 16px 16px 0 16px;
  grid-column: 1/3;
  padding: 8px 16px;
  font-size: ${rem(14)};
  color: ${state.danger};
  > span {
    margin-right: 4px;
  }
  > a {
    margin-left: 4px;
    text-decoration: underline;
    cursor: pointer;
  }
`;

export default AlertDanger;
