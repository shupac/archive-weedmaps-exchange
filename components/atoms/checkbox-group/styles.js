import styled from 'styled-components';
import theme from 'lib/styles/theme';

export const Row = styled.div`
  display: flex;
  align-items: center;
  margin-bottom: 14px;

  span {
    margin-left: 16px;
    font-size: 14px;
    color: ${theme.colors.oxfordBlue};
    user-select: none;
  }
`;

export default { Row };
