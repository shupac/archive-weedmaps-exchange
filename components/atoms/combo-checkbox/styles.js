import styled from 'styled-components';
import theme from 'lib/styles/theme';

export const Container = styled.div`
  width: 16px;
  height: 16px;
  border-radius: 3px;
  border: 1px solid;
  border-color: ${({ checked }) =>
    checked ? theme.colors.primary : theme.colors.border};
  background-color: ${({ checked }) =>
    checked ? theme.colors.primary : 'initial'};
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const Row = styled.div`
  display: flex;
  align-items: center;
  margin-bottom: 12px;

  span {
    margin-left: 16px;
    font-size: 14px;
    color: ${theme.colors.oxfordBlue};
    user-select: none;
  }
`;

export const PartialCheck = styled.div`
  height: 1.6px;
  width: 8px;
  border-radius: 1px;
  background-color: white;
`;
