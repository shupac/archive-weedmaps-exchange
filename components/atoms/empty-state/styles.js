// @flow
import WmTheme from '@ghostgroup/ui.theme';
import styled from 'styled-components';
import { rem } from 'polished';

const { text } = WmTheme.style;

export const NoResults = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 80px 0;
  width: 100%;
  grid-column: 1/-1;
  text-align: center;
  color: ${text.normal};
  > img {
    align-self: center;
  }
`;

export const NoResultsTitle = styled.h2`
  font-size: ${rem(20)};
  font-weight: 600;
`;

export const NoResultsInstructions = styled.span`
  font-size: ${rem(14)};
  margin-bottom: 24px;
  max-width: 500px;
  line-height: 20px;
  white-space: pre-line;
  overflow-wrap: break-word;
  text-align: center;
`;
