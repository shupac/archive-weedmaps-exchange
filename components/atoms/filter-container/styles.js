import styled from 'styled-components';
import { rem } from 'polished';
import theme from 'lib/styles/theme';

export const Container = styled.div`
  width: 100%;
  background-color: white;
`;

export const Header = styled.div`
  height: 64px;
  padding: 8px 16px;
  display: flex;
  align-items: center;
`;

export const FilterInfo = styled.div`
  display: flex;
  flex-direction: column;
  margin-right: auto;
`;

export const FilterName = styled.div`
  color: #354052;
  font-size: ${rem(16)};
  font-weight: 600;
  line-height: ${rem(24)};
  margin-top: -4px;
`;

export const FiltersLabel = styled.div`
  color: ${theme.colors.gullGray};
  font-size: ${rem(12)};
  line-height: ${rem(14)};
`;

export const Chevron = styled.div`
  cursor: pointer;
  transform: rotateZ(${({ collapsed }) => (collapsed ? '0deg' : '180deg')});
  transform-origin: center;
  transition: transform 0.3s;
`;

export const Expandable = styled.div`
  margin-right: 4px;
  overflow: ${({ contentHeight, maxHeight }) =>
    contentHeight > maxHeight ? 'auto' : 'hidden'};
  height: ${({ collapsed, contentHeight, maxHeight }) =>
    collapsed ? 0 : Math.min(maxHeight, contentHeight)}px;
  transition: height 0.3s;
  &::-webkit-scrollbar {
    width: 5px;
  }

  &::-webkit-scrollbar-thumb {
    background-color: ${theme.colors.divider};
    border-radius: 2.5px;
  }
`;

export const Filters = styled.div`
  padding: 0 16px;
  padding-bottom: 4px;
`;
