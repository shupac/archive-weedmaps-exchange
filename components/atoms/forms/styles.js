import styled, { css } from 'styled-components';
import { rem } from 'polished';
import theme from 'lib/styles/theme';

const InputStyles = css`
  font-size: ${rem(14)};
  font-weight: 300;
  width: ${props => props.width || '100%'};
  height: ${props => props.height || 40}px;
  border: 1px ${theme.style.state.secondaryCompanion} solid;
  border-radius: 3px;
  padding: 0 16px;
  background-color: ${({ disabled }) =>
    disabled ? theme.palette.lightGrey3 : theme.style.background.light};
  -webkit-appearance: none;
  ${({ hasError }) => hasError && `border-color: ${theme.style.state.danger}`};
  &::placeholder {
    color: ${theme.style.state.secondaryCompanion};
    font-style: italic;
  }
  &:focus {
    border-color: ${theme.style.state.primary};
  }
`;

const AreaStyles = css`
  resize: none;
  font-size: ${rem(14)};
  font-weight: 300;
  width: 100%;
  height: ${props => props.height || 72}px;
  border: 1px ${theme.style.state.secondaryCompanion} solid;
  border-radius: 3px;
  padding: 16px;
  background-color: ${({ disabled }) =>
    disabled ? theme.palette.lightGrey3 : theme.style.background.light};
  -webkit-appearance: none;
  ${({ hasError }) => hasError && `border-color: ${theme.style.state.danger}`};
  &::placeholder {
    color: ${theme.style.state.secondaryCompanion};
    font-style: italic;
  }
  &:focus {
    border-color: ${theme.style.state.primary};
  }
`;

AreaStyles.defaultProps = {
  hasError: false,
};

InputStyles.defaultProps = {
  hasError: false,
};

export const StyledInput = styled.input`
  ${InputStyles};
`;

export const StyledArea = styled.textarea`
  ${AreaStyles};
`;

export const InputWrap = styled.div`
  position: relative;
`;

export const InputError = styled.small`
  font-size: ${rem(11)};
  color: ${theme.style.state.danger};
`;
