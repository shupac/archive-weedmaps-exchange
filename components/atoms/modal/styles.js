import styled from 'styled-components';
import WmTheme from '@ghostgroup/ui.theme';
import { rem } from 'polished';

const { text, state, icon } = WmTheme.style;

export const ModalContainer = styled.div`
  background-color: ${state.light};
  border-radius: 3px;
  position: relative;
  pointer-events: auto;
  min-width: 200px;
  color: ${text.normal};
`;

export const CloseButton = styled.a`
  position: absolute;
  cursor: pointer;
  top: 16px;
  right: 16px;
  fill: ${icon.dark};
  z-index: 2;
`;

export const ModalHeader = styled.div`
  border-bottom: 1px solid ${icon.inverted};
  padding: 16px;
  font-size: ${rem(20)};
  font-weight: 600;
  margin: 0;
  user-select: none;
`;

export const ModalContent = styled.div`
  overflow: auto;
`;

export const ModalContentWrapper = styled.div`
  width: 620px;
  padding: 24px;
  color: ${text.normal};
  font-size: ${rem(14)};
`;

export const ButtonRow = styled.div`
  display: flex;
  justify-content: flex-end;
  margin-top: 24px;

  > button {
    width: auto;
    min-width: 132px;
    margin-right: 16px;
  }

  > :last-child {
    margin-right: 0;
  }
`;
