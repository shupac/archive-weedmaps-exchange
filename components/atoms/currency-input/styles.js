import styled from 'styled-components';
import { rem } from 'polished';
import WmTheme from '@ghostgroup/ui.theme';
import { InputStyles } from 'components/atoms/text-input/styles';

const { background, state } = WmTheme.style;

export const DollarSignWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: ${rem(14)};
  height: ${props => props.height || 40}px;
  background-color: ${background.secondary};
  width: 30px;
  border: 1px ${state.secondaryCompanion} solid;
  border-right: 0;
  border-radius: 3px 0 0 3px;
`;

export const Input = styled.input`
  ${InputStyles};
  border-radius: 0 3px 3px 0;
`;
