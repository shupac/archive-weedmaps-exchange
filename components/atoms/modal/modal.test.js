/* eslint-disable import/first */
/* globals document window */
import React from 'react';
import { shallow, mount } from 'enzyme';
import UiStore from 'lib/data-access/stores/ui';
import { Modal } from './';
import { ModalHeader } from './styles';

function setup() {
  const mockUiStore = UiStore.create();
  const mockStore = { uiStore: mockUiStore };
  const component = <Modal store={mockStore} />;
  const wrapper = shallow(component, {
    disableLifecycleMethods: true,
  });
  return { wrapper, mockStore };
}

describe('Modal', () => {
  it('should render a modal ', () => {
    const { wrapper } = setup();
    expect(wrapper.exists()).toEqual(true);
  });

  it('should be able to lock scrolling ', () => {
    const { wrapper } = setup();
    const instance = wrapper.instance();
    instance.lockBgScroll(true);
    const doc = [...document.body.classList];
    expect(doc[0]).toEqual('modal-open');
  });

  it('should be able to unlock scrolling ', () => {
    const { wrapper } = setup();
    const instance = wrapper.instance();
    instance.lockBgScroll(false);
    const doc = [...document.body.classList];
    expect(doc[0]).toBe(undefined);
  });

  it('should reset modal state when unmounts ', () => {
    const { wrapper, mockStore } = setup();
    const instance = wrapper.instance();
    const closeModal = jest.spyOn(mockStore.uiStore, 'closeModal');
    instance.componentWillUnmount();
    expect(closeModal).toHaveBeenCalled();
    closeModal.mockRestore();
  });

  it('should set the ref node', () => {
    const { wrapper } = setup();
    const instance = wrapper.instance();
    instance.setRef('node');
    expect(instance.nodeRef).toEqual('node');
  });

  it('should handle keydown events', () => {
    const mockUiStore = UiStore.create();
    const mockStore = { uiStore: mockUiStore };
    const keyDownHandler = jest.fn();
    const component = (
      <Modal store={mockStore} keyDownHandler={keyDownHandler} />
    );
    const wrapper = mount(component, {
      disableLifecycleMethods: true,
    });
    const instance = wrapper.instance();
    const onKeyDown = jest.spyOn(instance, 'onKeyDown');
    const keydownEvent = document.createEvent('Event');
    keydownEvent.initEvent('keydown', true, true);
    document.dispatchEvent(keydownEvent);
    expect(onKeyDown).toHaveBeenCalled();
    expect(keyDownHandler).toHaveBeenCalled();
    onKeyDown.mockRestore();
  });

  it('should handle mousedown events', () => {
    const mockUiStore = UiStore.create();
    const mockStore = { uiStore: mockUiStore };
    const mouseDownHandler = jest.fn();
    const component = (
      <Modal store={mockStore} mouseDownHandler={mouseDownHandler} />
    );
    const wrapper = mount(component, {
      disableLifecycleMethods: true,
    });
    const instance = wrapper.instance();
    instance.nodeRef = document;
    const onMouseDown = jest.spyOn(instance, 'onMouseDown');
    const mousedownEvent = document.createEvent('Event');
    mousedownEvent.initEvent('mousedown', true, true);
    document.dispatchEvent(mousedownEvent);
    expect(onMouseDown).toHaveBeenCalled();
    expect(mouseDownHandler).toHaveBeenCalled();
    onMouseDown.mockRestore();
  });

  it('should close modal on escape key', () => {
    const { wrapper, mockStore } = setup();
    const instance = wrapper.instance();
    const closeModal = jest.spyOn(mockStore.uiStore, 'closeModal');
    instance.onKeyDown({ keyCode: 27 });
    expect(closeModal).toHaveBeenCalled();
    closeModal.mockRestore();
  });

  it('will lock the background if has new props', async () => {
    const { wrapper } = setup();
    const instance = wrapper.instance();
    const lockBackgroundScroll = jest.spyOn(instance, 'lockBgScroll');
    const prevUiStore = UiStore.create({ activeModal: 'foo' });
    const store = { uiStore: prevUiStore };
    instance.componentDidUpdate({ store });
    expect(lockBackgroundScroll).toHaveBeenCalled();
    lockBackgroundScroll.mockRestore();
  });

  it('will not lock the background if props have not changed', async () => {
    const mockUiStore = UiStore.create({ activeModal: 'foo' });
    const mockStore = { uiStore: mockUiStore };
    const component = <Modal store={mockStore} />;
    const wrapper = mount(component, {
      disableLifecycleMethods: true,
    });
    const instance = wrapper.instance();
    const lockBackgroundScroll = jest.spyOn(instance, 'lockBgScroll');
    instance.componentDidUpdate({ store: mockStore });
    expect(lockBackgroundScroll).not.toHaveBeenCalled();
    lockBackgroundScroll.mockRestore();
  });

  it('will render the header if defined', () => {
    const mockUiStore = UiStore.create({ activeModal: 'foo' });
    const mockStore = { uiStore: mockUiStore };
    const component = <Modal store={mockStore} header="bar" />;
    const wrapper = mount(component, {
      disableLifecycleMethods: true,
    });
    expect(wrapper.find(ModalHeader).exists()).toEqual(true);
  });
});
