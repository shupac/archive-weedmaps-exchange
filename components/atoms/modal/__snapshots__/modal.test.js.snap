// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should be able to not render modal  1`] = `null`;

exports[`Modal should render a modal  1`] = `null`;
